src/networking.c

```diff
@@ -2927,6 +2927,8 @@ void clientCommand(client *c) {
 "    Report tracking status for the current connection.",
 "NO-EVICT (ON|OFF)",
```

Comment:
Another thing that I'm noticing while looking at this. We now have 3 "flag" commands:
* No-evict
* stealth-mode
* Reply

Note: Caching is not here, since it's a transitive flag. Tracking has a lot of weird state, so didn't seem appropriate to add here.

Maybe instead of adding a new subcommand, we make a command for setting flags. Something like:
```
CLIENT SETFLAG NO-EVICT YES STEALTH-MODE YES
```

I also think this should get added to HELLO. Then there can be clear parity between HELLO and CLIENT.

Reply:
Maybe also `CLIENT GETFLAG`?

Reply:
Flags should be present in `CLIENT LIST` I would imagine.

Reply:
I guess we still have to preseve `CLIENT NO-EVICT` and `CLIENT REPLY`, since there might be users using these two commands. But this is making the commands so redundant. Or perhaps we may deprecate `CLIENT NO-EVICT` and `CLIENT REPLY` someday?

I agree the idea of merging three subcommands as one, but for now it still looks fine for me.

Reply:
@oranagra @yossigo @itamarhaber Thoughts about introducing a new `CLIENT SETFLAGS` API instead of adding yet another client command. We can deprecate the other two and then add parity in client hello?

Reply:
i'm not certain unifying these into a common sub-command is right.
when we look at this from the redis internals side, we see these as flags, but it shouldn't necessarily reflect on the interface from the client, these 3 features are completely unrelated, and are not even likely to be used together.
i see the advantage of being able to set them all with a single sub-command and not needing to send 3 separate commands, but i think it doesn't justify re-designing the CLIENT command and making it an even more messed up than what it already is (due to supporting both the old and new APIs).

i do agree we want to expose these in HELLO, but there the problem is different, since we just need to add more argument for a command that already takes many optional arguments (unlike the CLIENT sub-commands), so it's easy.

p.s. think i'm the one who suggested a STEALTH mode client flag to resolve the LRU problem (in #11409), but looking at it now, i think the "stealth" mode feature can be even broader. i.e. a "stealth" client may also want to hide itself from commandstats, MONITOR, and maybe even SLOWLOG.
if that's the case, maybe CLIENT STEALTH is not a boolean thing, but an enum or a set of separate flags?

Reply:
from another point of view, `CLIENT SETFLAGS` may mess up ACL rules, for example we just wanna allow a user to off 
client's reply but don't wanna allow the user to flag client as no-evict.

so, I thinks it's better to keep the current subcommands.

Reply:
If we don't like the word STEALTH for this, then how about NOTOUCH? It is more to the point.

Reply:
@oranagra @soloestoy The ACL problem will also exist for HELLO. I think if we proceed during the current path, we're basically locking ourselves out of a consistent strategy for HELLo and CLIENT.

> when we look at this from the redis internals side, we see these as flags, but it shouldn't necessarily reflect on the interface from the client, these 3 features are completely unrelated, and are not even likely to be used together.

It's not actually relevant if they will be used together, the point is that they expose the same types of controls, namely that they modulate client behavior. Ideally there would be an ACL control for which configs they are able to set. I think blindly adding this is making the system harding to maintain long term.

Reply:
what i wanted to argue is that these 3 can remain separate sub-commands in CLIENT, and that only the STEALTH or NO-TOUCH one would be added to HELLO. arguably the NO-TOUCH isn't exposing any ACL concern.

consider a case where STEALTH has several modes NORMAL, HIDDEN, NOTOUCH, it could be exposed by a specific CLIENT sub-command, and a specific HELLO argument (pair of token and value).
point is, i don't think it has to be a generic SETFLAG sub-command, but rather something specific, and that might solve the ACL concern as well.